lib/ts/recipe/thirdparty/providers/types.ts

```diff
@@ -61,7 +61,7 @@ export type CustomProviderConfig = {
     /*
      * Button Component
      */
-    buttonComponent?: FC<{ name: string }> | { new (props: { name: string }): React.Component<any, any> };
+    buttonComponent?: ProviderConfig["buttonComponent"];
```

Comment:
I'd prefer if this went the other way (`ProviderConfig` referencing this) since this is a user-facing type, so it's better to have the entire definition available here. Before doing this, please check if this is the type we use in user input configs (I'm not a 100% sure)